### PR TITLE
Add oras_utils.py module to push and pull artifacts

### PR DESCRIPTION
**Pull request by yashvardhannanavati**
Oras doesn't work well with multiple auths for the same registry in config.json. This commit also enhances set_registry_auths to function such that it is possible to replace entire config.json with provided auth instead of simply appending to the existing file.

Refers to CLOUDDST-28647

Signed-off-by: Yashvardhan Nanavati <yashn@bu.edu>

Assisted-by: Cursor

## Summary by Sourcery

Introduce an ORAS utility module for OCI artifact operations and enhance registry authentication handling.

New Features:
- Add `get_oras_artifact` and `push_oras_artifact` functions to manage OCI artifact pull and push using ORAS with optional registry authentication.

Enhancements:
- Extend `set_registry_auths` context manager with a `use_empty_config` option to replace Docker config entirely with provided credentials instead of merging with a template.

Tests:
- Add unit tests for ORAS operations covering pull and push scenarios including authentication, annotations, failure cases, and credential leak prevention.
- Add a parametrized test for `set_registry_auths` behavior when using an empty configuration.

### CODE REVIEW

File: iib/workers/tasks/oras_utils.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            return temp_dir
+        except Exception as e:
+            # Clean up temp directory on failure
+            if os.path.exists(temp_dir):
+                shutil.rmtree(temp_dir)
+            raise IIBError(f'Failed to pull OCI artifact {artifact_ref}: {e}')
+
+
+@instrument_tracing(span_name="workers.tasks.oras_utils.push_oras_artifact")
+def push_oras_artifact(
```
[the remark this answers is not in the document]
Comment (yashvardhannanavati) [reply iib/workers/tasks/oras_utils.py:55]: 1. Excellent catch! Modified the code to allow absolute paths.
2. We do log a successful push even if it doesn't have a return value. I think let's stick to returning nothing for this one.

[the remark this answers is not in the document]
Comment (ashwgit) [reply iib/workers/tasks/oras_utils.py:60]: sounds great.

**Review by ashwgit — APPROVED** [review]

**Comment by yashvardhannanavati** [comment]
@sourcery-ai resolve
